core/lib/instrument.c

```diff
@@ -1791,7 +1791,7 @@ create_and_initialize_module_data(app_pc start, app_pc end, app_pc entry_point,
             copy->segments[i].end = os_segments[i].end;
             copy->segments[i].prot = os_segments[i].prot;
         }
-    } else
+    } else if (segments != NULL)
```

Comment:
Probably it's only required to keep the compiler happy. We could also put this at the top of create_and_initialize_module_data():

CLIENT_ASSERT((os_segments == NULL) != (segments == NULL)); /* Exactly one of the two is non-NULL. */